grpc/src/main/java/com/linecorp/armeria/server/grpc/UnframedGrpcErrorHandler.java

```diff
@@ -82,14 +122,36 @@ static UnframedGrpcErrorHandler ofJson() {
      * Returns a plain text response.
      */
     static UnframedGrpcErrorHandler ofPlainText() {
+        return ofPlainText(UnframedGrpcStatusMappingFunction.of());
+    }
+
+    /**
+     * Returns a plain text response.
+     *
+     * @param statusMappingFunction The function which maps the {@link Throwable} or gRPC {@link Status} code
+     *                              to an {@link HttpStatus} code.
+     */
+    static UnframedGrpcErrorHandler ofPlainText(UnframedGrpcStatusMappingFunction statusMappingFunction) {
+        // Ensure that unframedGrpcStatusMappingFunction never returns null
+        // by falling back to the default.
+        final UnframedGrpcStatusMappingFunction mappingFunction =
+                requireNonNull(statusMappingFunction, "statusMappingFunction")
+                        .orElse(UnframedGrpcStatusMappingFunction.of());
         return (ctx, status, response) -> {
             final Code grpcCode = status.getCode();
             final StringBuilder message = new StringBuilder("grpc-code: " + grpcCode.name());
             final String grpcMessage = status.getDescription();
             if (grpcMessage != null) {
                 message.append(", ").append(grpcMessage);
             }
-            final HttpStatus httpStatus = GrpcStatus.grpcStatusToHttpStatus(status);
+            final RequestLogAccess log = ctx.log();
+            final Throwable cause;
+            if (log.isAvailable(RequestLogProperty.RESPONSE_CAUSE)) {
+                cause = log.partial().responseCause();
```

Comment:
It seems like we can't get the `responseCause` because `RequestLogProperty.RESPONSE_CAUSE` is not set to the `flags` when `responseContent` is called.
So I think we should update the flag here. https://github.com/line/armeria/blob/master/core/src/main/java/com/linecorp/armeria/common/logging/DefaultRequestLog.java#L1268
Could you also add a test case that covers this line, please?

Reply:
Thank you for catching it :) Updated.
Anyway, just curious, is there any reason why not update this flag before?

Reply:
I think it was a bug. 😅 Sorry about it. 😅